RecoTauTag/RecoTau/plugins/TauDiscriminationAgainstElectronDeadECAL.cc

```diff
@@ -21,51 +20,36 @@
 #include "RecoTauTag/RecoTau/interface/TauDiscriminationProducerBase.h"
 #include "RecoTauTag/RecoTau/interface/AntiElectronDeadECAL.h"
 
-// helper function retrieve the correct tau type name
-template <class TauType>
-std::string getTauTypeNameString(bool capitalise) {
-  // this generic one shoudl never be called.
-  throw cms::Exception("TauDiscriminationAgainstElectronDeadECALBase")
-      << "Unsupported TauType used. You must use either reco::PFTau or pat::Tau.";
-}
-// template specialiazation to get the correct (PF/PAT)Tau type names
-template <>
-std::string getTauTypeNameString<reco::PFTau>(bool capitalise) {
-  return capitalise ? "PFRecoTau" : "pfRecoTau";
-}
-template <>
-std::string getTauTypeNameString<pat::Tau>(bool capitalise) {
-  return capitalise ? "PATTau" : "patTau";
-}
-
 template <class TauType, class TauDiscriminator>
-class TauDiscriminationAgainstElectronDeadECALBase : public TauDiscriminationProducerBase<TauType, TauDiscriminator> {
+class TauDiscriminationAgainstElectronDeadECAL : public TauDiscriminationProducerBase<TauType, TauDiscriminator> {
 public:
   typedef std::vector<TauType> TauCollection;
   typedef edm::Ref<TauCollection> TauRef;
-  explicit TauDiscriminationAgainstElectronDeadECALBase(const edm::ParameterSet& cfg)
+  explicit TauDiscriminationAgainstElectronDeadECAL(const edm::ParameterSet& cfg)
       : TauDiscriminationProducerBase<TauType, TauDiscriminator>::TauDiscriminationProducerBase(cfg),
         moduleLabel_(cfg.getParameter<std::string>("@module_label")),
         verbosity_(cfg.getParameter<int>("verbosity")),
         antiElectronDeadECAL_(cfg) {}
-  ~TauDiscriminationAgainstElectronDeadECALBase() override {}
+  ~TauDiscriminationAgainstElectronDeadECAL() override {}
 
   void beginEvent(const edm::Event& evt, const edm::EventSetup& es) override { antiElectronDeadECAL_.beginEvent(es); }
 
   double discriminate(const TauRef& tau) const override {
     if (verbosity_) {
-      edm::LogPrint(getTauTypeNameString<TauType>(true) + "AgainstEleDeadECAL")
-          << "<" + getTauTypeNameString<TauType>(true) + "AgainstElectronDeadECAL::discriminate>:";
-      edm::LogPrint(getTauTypeNameString<TauType>(true) + "AgainstEleDeadECAL") << " moduleLabel = " << moduleLabel_;
-      edm::LogPrint(getTauTypeNameString<TauType>(true) + "AgainstEleDeadECAL")
+      edm::LogPrint(TauDiscriminationProducerBase<TauType, TauDiscriminator>::getTauTypeString() + "AgainstEleDeadECAL")
```

Comment:
I'm a bit surprised to see the prefix.
I was expecting that ```getTauTypeString()``` alone should work

Reply:
```using TauDiscriminationProducerBase<TauType, TauDiscriminator>::getTauTypeString;```  somewhere in the scope of ```TauDiscriminationAgainstElectronDeadECAL``` should allow to avoid the lengthy prefix

Reply:
No it doesn't work alone without the prefix (compiler doesn't know actual type of the method), `using TauDiscriminationProducerBase<TauType, TauDiscriminator>::getTauTypeString;` doesn't help, too. The trick can be use of `this->getTauTypeString()`, which is more compact, so I adopt this.